### ISSUE TITLE
Bug: False positive with no-missing-label-refs on checkboxes

### ISSUE THREAD

**Issue by chad-loder** [eslint/markdown#335]
### Environment

ESLint version: 9.23.0
@eslint/markdown version: 6.3.0
Node version: v22.14.0
npm version: 10.9.2
Operating System: MacOS Sequoia

### Which language are you using?

gfm

### What did you do?

<details>
<summary>Configuration</summary>

```javascript
// Custom markdown rules for content and formatting
const markdownContentRules = {
  // Warnings for style/formatting issues
  'markdown/fenced-code-language': SEVERITY.ERROR, // Error about missing code languages
  'markdown/heading-increment': SEVERITY.WARN, // Warn about skipped heading levels
  'markdown/no-duplicate-headings': SEVERITY.WARN, // Warn about duplicate headings

  // Errors for content/usability issues
  'markdown/no-empty-links': SEVERITY.ERROR, // Error on empty links
  'markdown/no-missing-label-refs': SEVERITY.ERROR, // Error on missing refs
  'markdown/no-invalid-label-refs': SEVERITY.ERROR, // Error on invalid refs

  // Other rules
  'markdown/no-html': SEVERITY.OFF, // Allow HTML in markdown
};
```
</details>

Using this markdown code for checkboxes:

```markdown
## Critical Issues

- [ ] This is an unchecked item
- [x] This is a checked item
```

### What did you expect to happen?

No warnings/errors

### What actually happened?

```json
[{
	"resource": "test.md",
	"owner": "eslint",
	"code": "markdown/no-missing-label-refs",
	"severity": 8,
	"message": "Label reference '' not found.",
	"source": "eslint",
}]
[{
	"resource": "test.md",
	"owner": "eslint",
	"code": "markdown/no-missing-label-refs",
	"severity": 8,
	"message": "Label reference 'x' not found.",
	"source": "eslint",
}]
```

### Link to Minimal Reproducible Example

https://gist.github.com/chad-loder/73d61eb898b29a5852209145acfef4f0

### Participation

- [x] I am willing to submit a pull request for this issue.

### Additional comments

The checkbox is part of [Markdown extended syntax](https://www.markdownguide.org/extended-syntax/). I have tried with both GFM and regular markdown language options in my linter settings. It's possible that I am not configuring by language option correctly, in which case I apologize.

**Comment by nzakas** [comment]
Thanks for the report. Right now, we don't support extended syntax. I'm not yet sure how best to handle this, so I'll leave this open until I have time to circle back.

**Comment by nzakas** [comment]
@lumirlumir thanks for pointing this out. I confirmed here:

https://stackblitz.com/edit/stackblitz-starters-nnn59xks

(As a reminder to us all: This is why we need to insist on reproduction cases.)
